controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/AbstractRequestProcessor.java

```diff
@@ -66,8 +67,20 @@
 @Slf4j
 public abstract class AbstractRequestProcessor<T extends ControllerEvent> extends SerializedRequestHandler<T> 
         implements StreamRequestProcessor {
-    protected static final Predicate<Throwable> OPERATION_NOT_ALLOWED_PREDICATE = e -> Exceptions.unwrap(e) 
-            instanceof StoreException.OperationNotAllowedException;
+    protected static final Predicate<Throwable> ILLEGAL_STATE_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.IllegalStateException;
+    protected static final Predicate<Throwable> DATA_NOT_FOUND_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.DataNotFoundException;
+    protected static final Predicate<Throwable> SEGMENT_NOT_FOUND_PREDICATE = e -> Exceptions.unwrap(e) instanceof StoreException.DataContainerNotFoundException;
+    protected static final Predicate<Throwable> EVENT_RETRY_PREDICATE = ILLEGAL_STATE_PREDICATE.or(DATA_NOT_FOUND_PREDICATE)
```

Comment:
nit: lets rename it event_writeback_predicate.
also, if we write back the event, then presently we still fail the future with the cause for failure which goes to concurrenteventprocessor which also has a writeback logic on retryable exceptions. 
we should actually complete the future in `withCompletionMethod` (line 182)

Reply:
fixed.